lib/components/twitch/message.dart

```diff
@@ -59,7 +59,73 @@ Color lighten(Color color, [double amount = .1]) {
   return hslLight.toColor();
 }
 
-Iterable<InlineSpan> parseText(String text, TextStyle linkStyle) {
+dynamic tokenize(msg) {
```

Comment:
Instead of returning tuples, create separate class types to contain the metadata you want to return.